src/org/zaproxy/zap/spider/parser/SpiderHtmlFormParser.java

```diff
@@ -242,6 +252,19 @@ private FormData prepareFormDataSet(FormFields form) {
 				currentList = submitFields;
 			}
 			for (String value : getValues(field)) {
```

Comment:
We have moved the logic from the getValues method. Please let us know if this was done properly or needs to be revised.

At the moment defaultValues are no longer being used. It seems the logic within the for loop no longer works as we cannot call the getDefaultTextValue and return a type string. How would we go about passing the field into the generator again? Or should it already have been passed in the first time(should we restructure our loop)?